### ISSUE TITLE
[Feature Request] Block constraint for vertices

### ISSUE THREAD

**Issue by FEA-eng** [FreeCAD/FreeCAD#6831]
### Is there an existing request for this?

- [X] I have searched the existing requests

### Forums discussion

https://forum.freecadweb.org/viewtopic.php?f=3&t=68434

### Subproject(s) affected?

Sketcher

### Idea description

Currently, the block constraint doesn't work for free points (reported here: #5844) but it would also be good to make it work for vertices (points belonging to other geometrical objects).

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by PaddleStroke** [comment]
Duplicate of https://github.com/FreeCAD/FreeCAD/issues/5844 it seems

**Comment by FEA-eng** [comment]
> Duplicate of #5844 it seems

Not really. I've mentioned #5844 in the description of this issue. The older one is about free points (Sketcher CreatePoint) while this one is about vertices of lines or other geometrical entities.

**Comment by PaddleStroke** [comment]
Ah sorry I have not seen you mentioned it already.

**Comment by AjinkyaDahale** [comment]
Isn't this (and #5844) just the lock constraint?

**Comment by FEA-eng** [comment]
> Isn't this (and #5844) just the lock constraint?

Lock constraint only applies horizontal and vertical dimensions automatically (the same could be done manually). Block constraint doesn't apply any dimensional constraints and just makes the line immovable (helpful for debugging or quickly and temporarily constraining the sketch in general).

**Comment by AjinkyaDahale** [comment]
> Lock constraint only applies horizontal and vertical dimensions automatically (the same could be done manually). Block constraint doesn't apply any dimensional constraints and just makes the line immovable (helpful for debugging or quickly and temporarily constraining the sketch in general).

But for vertices this has the same effect.

**Comment by FEA-eng** [comment]
> But for vertices this has the same effect.

In both cases, a geometrical entity ends up fixed but Lock adds 2 editable dimension constraints to a vertex while Block is a single non-editable constraint applied to the whole line. If it was possible to apply it to vertices as well, there would be a quick way to fix multiple vertices in place, without adding any actual dimensions. Lock is basically just a mode of the new Dimension tool now - it automatically applies 2 dimension constraints linked to the origin without asking for values. Block is better for testing/debugging without cluttering the sketch with more dimensions.

SolidWorks also lets you fix a vertex in place without dimensioning it:

![fix](https://github.com/FreeCAD/FreeCAD/assets/59876896/7192496a-31a7-4665-aaed-d2bb9517088a)
